File: Conway's Game of Life/[VB]-Conway's Game of Life/VB/Conways Game of Life/program/gui/frmProgress.vb
Public Class frmProgress

    Public Sub New(ByVal max As Integer, ByVal caption As String)

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Me.Text = caption
        ProgressBar1.Maximum = max
    End Sub

    Public Sub performStep()
        ProgressBar1.PerformStep()
        Label1.Text = ((ProgressBar1.Value / ProgressBar1.Maximum) * 100).ToString & "% complete"
        Me.Refresh()
        Application.DoEvents()
        If ProgressBar1.Value = ProgressBar1.Maximum Then Me.Close()
    End Sub

End Class

File: Conway's Game of Life/[C#]-Conway's Game of Life/C#/Conways Game of Life cs/app/gui/frmProgress.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Conways_Game_of_Life_cs
{
    public partial class frmProgress : Form
    {
        public frmProgress(int max, string caption)
        {
            // This call is required by the Windows Form Designer.
            InitializeComponent();

            // Add any initialization after the InitializeComponent() call.
            this.Text = caption;
            ProgressBar1.Maximum = max;
        }

        public void performStep()
        {
            ProgressBar1.PerformStep();
            Label1.Text = ((ProgressBar1.Value / ProgressBar1.Maximum) * 100).ToString() + "% complete";
            this.Refresh();
            Application.DoEvents();
            if (ProgressBar1.Value == ProgressBar1.Maximum)
                this.Close();
        }
    }
}
